design/ceph/object/store.md

```diff
@@ -73,7 +74,124 @@ If `preservePoolsOnDelete` is set to 'true' the pools used to support the object
   preservePoolsOnDelete: true
 ```
 
-If there is a `zone` section in object-store configuration, then the pool section in the ceph-object-zone resource will be used to define the pools.
+#### Pools shared by multiple CephObjectStore
+
+If user want to use existing pools for metadata and data, the pools must be created before the object store is created. This will be useful if multiple objectstore can share same pools. The detail of pools need to shared in `radosNamespaces` settings in object-store CRD. Now the object stores can consume same pool isolated with different namespaces. User can create via [Pool CRD](/Documentation/CRDs/Block-Storage/ceph-block-pool-crd.md), this is need to present before the object store is created.
+
+```yaml
+  radosNamespaces:
+    metadataPoolName: rgw-meta-pool
+    dataPoolName: rgw-data-pool
+```
+
+To create the pools that will be shared by multiple object stores, create the following CephBlockPool CRs:
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-meta-pool
+  spec:
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+---
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+ metadata:
+    name: rgw-data-pool
+  spec:
+    failureDomain: osd
+      erasureCoded:
+        dataChunks: 6
+        codingChunks: 2
+    application: rgw
+```
+
+The `.rgw.root` pool must be defined before the object store is created. This pool is used for storing the root info for all object stores.
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPool
+metadata:
+    name: rgw-root
+  spec:
+    name: .rgw.root
+    failureDomain: host
+    replicated:
+      size: 3
+    parameters:
+      pg_num: 8
+    application: rgw
+```
+
+The pools for this configuration will be created as below:
+
+```bash
+# ceph osd pool ls
+.rgw.root
+rgw-meta-pool
+rgw-data-pool
+```
+
+And the pool configuration in zone is as below:
+
+```json
+# radosgw-admin zone get --rgw-zone=my-store
+{
+    "id": "2220eb5f-2751-4a51-9c7d-da4ce1b0e4e1",
+    "name": "my-store",
+    "domain_root": "rgw-meta-pool:my-store.root",
+    "control_pool": "rgw-meta-pool:my-store.control",
+    "gc_pool": "rgw-meta-pool:my-store.gc",
+    "lc_pool": "rgw-meta-pool:my-store.lc",
+    "log_pool": "rgw-meta-pool:my-store",
+    "intent_log_pool": "rgw-meta-pool:my-store.intent",
+    "usage_log_pool": "rgw-meta-pool:my-store.usage",
+    "roles_pool": "rgw-meta-pool:my-store.roles",
+    "reshard_pool": "rgw-meta-pool:my-store.reshard",
+    "user_keys_pool": "rgw-meta-pool:my-store.users.keys",
+    "user_email_pool": "rgw-meta-pool:my-store.users.email",
+    "user_swift_pool": "rgw-meta-pool:my-store.users.swift",
+    "user_uid_pool": "rgw-meta-pool:my-store.users.uid",
+    "otp_pool": "rgw-meta-pool:my-store.otp",
```

Comment:
It seems to me that it is probably best to try to keep good 1:1 matching between pools and namespaces where possible. I notice that in this design, some of the info that used to be codified by different pool names is lost. When there are separate pools used for separate "functions", that is easy to read. For example, there used to be a `log` pool, but `log` is no longer mentioned here with your proposed namespace layout.

I think that it would be best to make sure all of the information that was previously codified is retained. This will make it easier to understand comparisons between namespaced and non-namespaced object stores. I also think that it will more easily allow us to support more types of meta pools in the future if we need to expand the design (i.e., if we need to support shared control/log/index/non-ec/otp pools in the future).

For example, this is what the `default` zone/objectstore looks like from some Ceph docs AFAICT:
```
radosgw-admin zone get
{
    "id": "adacbe1b-02b4-41b8-b11d-0d505b442ed4",
    "name": "default",
    "domain_root": "default.rgw.meta:root",
    "control_pool": "default.rgw.control",
    "gc_pool": "default.rgw.log:gc",
    "lc_pool": "default.rgw.log:lc",
    "log_pool": "default.rgw.log",
    "intent_log_pool": "default.rgw.log:intent",
    "usage_log_pool": "default.rgw.log:usage",
    "roles_pool": "default.rgw.meta:roles",
    "reshard_pool": "default.rgw.log:reshard",
    "user_keys_pool": "default.rgw.meta:users.keys",
    "user_email_pool": "default.rgw.meta:users.email",
    "user_swift_pool": "default.rgw.meta:users.swift",
    "user_uid_pool": "default.rgw.meta:users.uid",
    "otp_pool": "default.rgw.otp",
    "system_key": {
        "access_key": "",
        "secret_key": ""
    },
    "placement_pools": [
        {
            "key": "default-placement",
            "val": {
                "index_pool": "default.rgw.buckets.index",
                "storage_classes": {
                    "STANDARD": {
                        "data_pool": "default.rgw.buckets.data"
                    }
                },
                "data_extra_pool": "default.rgw.buckets.non-ec",
                "index_type": 0
            }
        }
    ],
    "realm_id": "",
    "notif_pool": "default.rgw.log:notif"
}
```
I note that the pool/namespace format is `<zone>.rgw.<function>:<subfunction>`.

I think we should make sure we preserve all this info by rearranging it to this format when namespaced: `<shared-pool>:<zone>.<function>.<subfunction>`
```suggestion
    "domain_root": "rgw-meta-pool:my-store.meta.root",
    "control_pool": "rgw-meta-pool:my-store.control",
    "gc_pool": "rgw-meta-pool:my-store.log.gc",
    "lc_pool": "rgw-meta-pool:my-store.log.lc",
    "log_pool": "rgw-meta-pool:my-store.log",
    "intent_log_pool": "rgw-meta-pool:my-store.log.intent",
    "usage_log_pool": "rgw-meta-pool:my-store.log.usage",
    "roles_pool": "rgw-meta-pool:my-store.meta.roles",
    "reshard_pool": "rgw-meta-pool:my-store.log.reshard",
    "user_keys_pool": "rgw-meta-pool:my-store.meta.users.keys",
    "user_email_pool": "rgw-meta-pool:my-store.meta.users.email",
    "user_swift_pool": "rgw-meta-pool:my-store.meta.users.swift",
    "user_uid_pool": "rgw-meta-pool:my-store.meta.users.uid",
    "otp_pool": "rgw-meta-pool:my-store.otp",
```

I guess this would also mean changing some below content as well, like `rgw-meta-pool:my-store.buckets.index`, `rgw-data-pool:my-store.buckets.data`, `rgw-meta-pool:my-store.buckets.non-ec`, and `rgw-meta-pool:my-store.log.notif`.

This is assuming that index and non-ec are supposed to use configs for the meta pool.